compatible ids (#1468)
conan-io/docs: reference/conanfile/methods.rst

Same behavior applies if you use the deprecated setting ``cppstd``.

CompatiblePackage
^^^^^^^^^^^^^^^^^
The ``package_id()`` method serves to define the "canonical" binary package ID, the identifier of the binary that correspond to the
input configuration of settins and options. This canonical binary package ID will be always computed, and Conan will check for its
existence to be downloaded and installed.

> If the binary of that package ID is not found, Conan allows to define an ordered list of compatible package IDs, of other configurations

Reviewer: ```suggestion
If the binary of that package ID is not found, Conan lets the recipe writer define an ordered list of compatible package IDs, of other configurations
```

> that are fully binary compatible and can be used as a fallback. The syntax to do this is:

Reviewer: ```suggestion
that should be binary compatible and can be used as a fallback. The syntax to do this is:
```

.. code-block:: python

    from conans import ConanFile, CompatiblePackage

    class Pkg(ConanFile):
        settings = "os", "compiler", "arch", "build_type"

        def package_id(self):
            if self.settings.compiler == "gcc" and self.settings.compiler.version == "4.9":
                compatible_pkg = CompatiblePackage(self)
                compatible_pkg.settings.compiler.version = "4.8"
                self.compatible_packages.append(compatible_pkg)

> This will define that if we try to install this package with ``gcc 4.9`` and there is not binary for that configuration, it will check

Reviewer: ```suggestion
This will define that, if we try to install this package with ``gcc 4.9`` and there isn't a binary available for that configuration, Conan will check
```

> if there is an available binary built with ``gcc 4.8`` and use it if existing. But not the other way round.

Reviewer: ```suggestion
if there is one available built with ``gcc 4.8`` and use it. But not the other way round.
```

.. seealso::

>     For more information about :ref:`CompatiblePackage read this <compatible_package>`

Reviewer: ```suggestion
    For more information about :ref:`CompatiblePackage read this <compatible_packages>`
```

Same behavior applies if you use the deprecated setting ``cppstd``.

CompatiblePackage
^^^^^^^^^^^^^^^^^
The ``package_id()`` method serves to define the "canonical" binary package ID, the identifier of the binary that correspond to the
input configuration of settins and options. This canonical binary package ID will be always computed, and Conan will check for its
existence to be downloaded and installed.

If the binary of that package ID is not found, Conan allows to define an ordered list of compatible package IDs, of other configurations
that are fully binary compatible and can be used as a fallback. The syntax to do this is:

.. code-block:: python

    from conans import ConanFile, CompatiblePackage

    class Pkg(ConanFile):

>         settings = "os", "compiler", "arch", "build_type"

Reviewer 2: ```suggestion
        settings = "os", "compiler", "arch", "build_type"

```

        def package_id(self):
            if self.settings.compiler == "gcc" and self.settings.compiler.version == "4.9":
                compatible_pkg = CompatiblePackage(self)
                compatible_pkg.settings.compiler.version = "4.8"
                self.compatible_packages.append(compatible_pkg)

This will define that if we try to install this package with ``gcc 4.9`` and there is not binary for that configuration, it will check
if there is an available binary built with ``gcc 4.8`` and use it if existing. But not the other way round.

> 

Reviewer 2: ```suggestion
```

.. seealso::

    For more information about :ref:`CompatiblePackage read this <compatible_package>`

> 

Reviewer 2: ```suggestion
```
